jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -0,0 +1,415 @@
+        }
+    }
+
+    /**
+     * Set the default generator options.
+     *
+     * @param configBuilder The builder on which to set the options
+     */
+    private void setDefaultOptions(SchemaGeneratorConfigBuilder configBuilder) {
+        configBuilder.with(Option.DEFINITIONS_FOR_ALL_OBJECTS);
+    }
+
+    /**
+     * Set the generator options form the configuration.
+     *
+     * @param configBuilder The configbuilder on which the options are set
+     * @param options       The options from the pom file
+     * @throws MojoExecutionException An exception in case of unexpected behavior
+     */
+    private void setOptionsFromConfig(SchemaGeneratorConfigBuilder configBuilder, String[] options) throws MojoExecutionException {
+        for (String option : options) {
+            try {
+                Option optionLiteral = Option.valueOf(option);
+                configBuilder = configBuilder.with(optionLiteral);
+            } catch (IllegalArgumentException e) {
+                throw new MojoExecutionException("Error: Unknown option " + option, e);
+            }
+        }
+    }
+
+    /**
+     * Configure all the modules on the generator.
+     *
+     * @param configBuilder The builder on which the modules are added.
+     * @param modules       The modules configuration part from the the pom.xml
+     * @throws MojoExecutionException Error exception
+     */
+    private void configureModules(SchemaGeneratorConfigBuilder configBuilder, GeneratorModule[] modules) throws MojoExecutionException {
+        for (GeneratorModule module : modules) {
+            if (module.className != null && !module.className.isEmpty()) {
+                try {
+                    getLog().info("- Adding CustomModule " + module.className);
+                    Class<? extends Module> moduleClass =
+                            (Class<? extends Module>) getClassLoader().loadClass(module.className);
+                    Module moduleInstance = moduleClass.getConstructor().newInstance();
+                    configBuilder.with(moduleInstance);
+                } catch (ClassCastException | InstantiationException
+                        | IllegalAccessException | NoSuchMethodException
+                        | InvocationTargetException | ClassNotFoundException e) {
+                    throw new MojoExecutionException("Error: Can not create custom module" + module.className, e);
+                }
+            } else if (module.name != null) {
+                switch (module.name) {
+                case "Jackson":
+                    getLog().info("- Adding JacksonModule");
+                    addJacksonModule(configBuilder, module);
+                    break;
+                case "JavaxValidation":
+                    getLog().info("- Adding Javax Validation Module");
+                    addJavaxValidationModule(configBuilder, module);
+                    break;
+                case "Swagger":
```

Comment:
There is a big difference between Swagger 1.5.x (for which there is a module already) and Swagger 2.x (now the OpenAPI standard, for which there is no module yet but will most likely come in the future).

Please include the 1.5 here as well, to avoid breaking or confusing changes when the other Swagger module is being added.

Reply:
ok